File: app/src/main/java/be/hcpl/android/filmtag/AboutFragment.java
package be.hcpl.android.filmtag;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.Html;
import android.text.util.Linkify;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.io.BufferedReader;
import java.io.Closeable;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import be.hcpl.android.filmtag.template.TemplateFragment;

/**
 * Created by hcpl on 6/08/15.
 */
public class AboutFragment extends TemplateFragment {

    private static final String SYSTEM_LINE_SEPARATOR = System.getProperty("line.separator");

    public static AboutFragment newInstance() {
        Bundle args = new Bundle();
        AboutFragment fragment = new AboutFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    protected int getLayoutResourceId() {
        return R.layout.fragment_about;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        String aboutText = readFile(getResources().openRawResource(R.raw.about));
        // add version
        aboutText = aboutText.replace("{version}", BuildConfig.VERSION_NAME);
        // set text
        TextView textView = ((TextView)view.findViewById(R.id.text_about));
        textView.setText(Html.fromHtml(aboutText));
        // and make clickable
        Linkify.addLinks(textView, Linkify.ALL);
    }

    // TODO move to utils instead
    private String readFile(InputStream stream) {
        BufferedReader in = null;

        try {
            in = new BufferedReader(new InputStreamReader(stream));
            StringBuilder buffer = new StringBuilder();

            String e;
            while((e = in.readLine()) != null) {
                buffer.append(e).append(SYSTEM_LINE_SEPARATOR);
            }

            StringBuilder var5 = buffer;
            return var5.toString();
        } catch (IOException var8) {
            ;
        } finally {
            closeStream(in);
        }

        return "";
    }

    private void closeStream(Closeable stream) {
        if(stream != null) {
            try {
                stream.close();
            } catch (IOException var2) {
                ;
            }
        }

    }
}

File: app/src/main/java/be/hcpl/android/filmtag/AboutFragment.kt
package be.hcpl.android.filmtag


import android.os.Bundle
import android.text.Html
import android.text.util.Linkify
import android.view.View
import android.widget.TextView
import androidx.fragment.app.Fragment

import java.io.BufferedReader
import java.io.Closeable
import java.io.InputStream
import java.io.InputStreamReader

/**
 * Created by hcpl on 6/08/15.
 */
class AboutFragment : Fragment(R.layout.fragment_about) {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        var aboutText = readFile(resources.openRawResource(R.raw.about))
        // add version
        aboutText = aboutText.replace("{version}", BuildConfig.VERSION_NAME)
        // set text
        val textView = view.findViewById(R.id.text_about) as? TextView
        textView?.apply {
            text = Html.fromHtml(aboutText)
            // and make clickable
            Linkify.addLinks(textView, Linkify.ALL)
        }
    }

    // TODO move to utils instead
    private fun readFile(stream: InputStream): String {
        var input: BufferedReader? = null

        try {
            input = BufferedReader(InputStreamReader(stream))
            val buffer = StringBuilder()

            var e: String? = input.readLine()
            while (e != null) {
                buffer.append(e).append(SYSTEM_LINE_SEPARATOR)
                e = input.readLine()
            }

            return buffer.toString()
        } catch (_: Exception) {
        } finally {
            closeStream(input)
        }

        return ""
    }

    private fun closeStream(stream: Closeable?) {
        if (stream != null) {
            try {
                stream.close()
            } catch (_: Exception) {
            }

        }

    }

    companion object {

        private val SYSTEM_LINE_SEPARATOR = System.lineSeparator()

    }
}
